stock_inventory_revaluation/models/stock_inventory_revaluation.py

```diff
@@ -0,0 +1,481 @@
+                            {'standard_price': revaluation.new_cost})
+                    else:
+                        revaluation.old_cost = revaluation.current_cost
+                        revaluation.old_value = revaluation.current_value
+                        value_diff = revaluation.current_value - \
+                            revaluation.new_value
+                        new_cost = value_diff / revaluation.qty_available
+                        revaluation.product_id.with_context(
+                            force_company=revaluation.company_id.id
+                        ).sudo().write({'standard_price': new_cost})
+            if revaluation.product_id.categ_id.\
+                    property_valuation == 'real_time':
+                revaluation.sudo()._create_accounting_entry()
+            self.post_date = fields.Datetime.now()
+            self.state = 'posted'
+
+            amount_diff = 0.0
+            if revaluation.product_id.\
+                    cost_method == 'real':
+                for reval_quant in revaluation.reval_quant_ids:
+                    amount_diff += reval_quant.get_total_value()
+                    reval_quant._write_new_cost()
+                if amount_diff == 0.0:
+                    return True
+            else:
+                if revaluation.product_id.\
+                        cost_method in ['standard', 'average']:
+                    if float_compare(revaluation.new_value, 0.0,
+                                     precision_rounding=account_prec) < 0:
+                        raise UserError(
+                            _("The new value for product %s cannot "
+                              "be negative" %
+                              revaluation.product_template_id.name))
+                    if float_compare(revaluation.qty_available, 0.0,
+                                     precision_rounding=product_prec) <= 0:
+                        raise UserError(
+                            _("Cannot do an inventory value change if the "
+                              "quantity available for product %s "
+                              "is 0 or negative" %
+                              revaluation.product_template_id.name))
+
+    @api.model
+    def create(self, values):
+        sequence_obj = self.env['ir.sequence']
+        if values.get('name', '/') == '/':
+            values['name'] = sequence_obj.next_by_code(
+                'stock.inventory.revaluation')
+        return super(StockInventoryRevaluation, self).create(values)
+
+    @api.multi
+    def button_post(self):
+        self.post()
+        return True
+
+    @api.multi
+    def button_draft(self):
+        self.write({'state': 'draft'})
+        return True
+
+    @api.multi
+    def button_cancel(self):
+        for revaluation in self:
+            for reval_quant in revaluation.reval_quant_ids:
+                reval_quant.quant_id.sudo().write(
+                    {'cost': reval_quant.old_cost})
```

Comment:
This only works for realtime inventory valuation? If inventory valuation is average shouldn't the cost price be reset?